integration/obscurogateway/obscurogateway_test.go

```diff
@@ -83,14 +85,18 @@ func TestObscuroGateway(t *testing.T) {
 	err := waitServerIsReady(httpURL)
 	require.NoError(t, err)
 
+	// prefunded wallet
+	w := wallet.NewInMemoryWalletFromConfig(genesis.TestnetPrefundedPK, integration.ObscuroChainID, testlog.Logger())
+
 	// run the tests against the exis
-	for name, test := range map[string]func(*testing.T, string, string){
+	for name, test := range map[string]func(*testing.T, string, string, wallet.Wallet){
```

Comment:
The wallet is known, there's no need to pass it as an argument. Have a look at `testAreTxsMinted` roughly line 110.

Reply:
if we create wallet instance in each test tests don't work as expected. (I assume nonce is not correct). Each test separately was ok. (I even tried changing the order and it didn't help).

Reply:
Makes sense 👍  we need to find a different funding approach, but something for the future perhaps.